go/fuzz_test.go

```diff
@@ -122,3 +155,49 @@ func FuzzCombineProofs(f *testing.F) {
 		_, _ = CombineProofs(proofs)
 	})
 }
+
+type verifyJSON struct {
+	Proof *CommitmentProof
+	Ref   *RefData
+	Spec  *ProofSpec
+}
+
+func FuzzVerifyMembership(f *testing.F) {
+	seeds := VectorsTestData()
+
+	// VerifyMembership requires:
+	//  Spec, ExistenceProof, CommitmentRootref,
+	for _, seed := range seeds {
+		proof, ref := LoadFile(f, seed.Dir, seed.Filename)
+		root, err := proof.Calculate()
+		if err != nil {
+			continue
+		}
+		if !bytes.Equal(ref.RootHash, root) {
+			continue
+		}
+
+		if ref.Value == nil {
+			continue
+		}
+
+		// Now serialize this value as a seed.
+		blob, err := json.Marshal(&verifyJSON{Proof: proof, Ref: ref, Spec: seed.Spec})
+		if err == nil {
+			f.Add(blob)
+		}
+	}
+
+	// 2. Now let's run the fuzzer.
+	f.Fuzz(func(t *testing.T, input []byte) {
+		var con verifyJSON
+		if err := json.Unmarshal(input, &con); err != nil {
+			return
+		}
+		spec, ref, proof := con.Spec, con.Ref, con.Proof
+		if ref == nil {
+			return
+		}
+		_ = VerifyMembership(spec, ref.RootHash, proof, ref.Key, ref.Value)
+	})
+}
```

Comment:
Wondering if we could also write this test like this:

```go
func FuzzVerifyMembership(f *testing.F) {
	if testing.Short() {
		f.Skip("in -short mode")
	}

	// 1. Add some seeds.
	for _, batchVec := range batchVectorDataSeeds {
		blob, err := json.Marshal(batchVec)
		if err != nil {
			f.Fatal(err)
		}
		f.Add(blob)
	}

	// 2. Now let's run the fuzzer.
	f.Fuzz(func(t *testing.T, input []byte) {
		var bv BatchVectorData
		if err := json.Unmarshal(input, &bv); err != nil {
			return
		}

		if bv.Ref == nil || bv.Proof == nil || bv.Ref.RootHash == nil {
			return
		}

		root, err := bv.Proof.Calculate()
		if err != nil {
			return
		}

		if !bytes.Equal(bv.Ref.RootHash, root) {
			return
		}

		_ = VerifyMembership(bv.Spec, bv.Ref.RootHash, bv.Proof, bv.Ref.Key, bv.Ref.Value)
	})
}
```

Because then it follows a similar structure as `FuzzVerifyNonMembership` and we don't need the extra `verifyJSON` type.

Reply:
See the difference here though is that I want to mutate on the already calculated Spec & Root; otherwise it'll take so much much longer to generate the respective values which is also how the prior fuzzer wasn't able to quickly find such values. Sure I can move it up though.